napari/components/viewer_model.py

```diff
@@ -364,6 +365,32 @@ def to_svg(self, file=None, view_box=None):
 
         return svg
 
+    def add_layer_by_type(self, layer_type, data, **kwargs):
+        """
+        Convenience method that maps a LayerType to it's add_layer method.
+
+        Parameters
+        ----------
+        layer_type : LayerTypes
+            Layer type to add
+        data :
+            The layer data to add
+        """
+        if layer_type is LayerTypes.LABELS:
+            return self.add_labels(data, **kwargs)
+        if layer_type is LayerTypes.POINTS:
+            return self.add_points(data, **kwargs)
+        if layer_type is LayerTypes.VECTORS:
+            return self.add_vectors(data, **kwargs)
+        if layer_type is LayerTypes.SHAPES:
+            return self.add_shapes(data, **kwargs)
+        if layer_type is LayerTypes.SURFACE:
+            return self.add_surface(data, **kwargs)
+        if layer_type is LayerTypes.IMAGE or layer_type is LayerTypes.VOLUME:
+            return self.add_image(data, **kwargs)
+        if layer_type == LayerTypes.PYRAMID:
+            return self.add_image(data, is_pyramid=True, **kwargs)
```

Comment:
As noted by the Enum, I'd drop `Pyramid` - in most cases we should do the right thing even if `is_pyramid` isn't set and I don't really think of it conceptually as it's own layer type anymore